projects/media-viewer/src/lib/media-viewer/annotation/annotation-store.service.ts

```diff
@@ -0,0 +1,269 @@
+import {Injectable, OnDestroy} from '@angular/core';
+import {HttpErrorResponse, HttpResponse} from '@angular/common/http';
+import {Observable, Subscription, Subject, BehaviorSubject, throwError} from 'rxjs';
+import {catchError} from 'rxjs/operators';
+import {v4 as uuid} from 'uuid';
+import {Annotation, Comment, IAnnotationSet} from './annotation-set.model';
+import {EmLoggerService} from '../../logging/em-logger.service';
+import {AnnotationApiHttpService} from './annotation-api-http.service';
+
+@Injectable()
+export class AnnotationStoreService implements OnDestroy {
+
+    annotationChangeSubscription: Subscription;
+    private commentBtnSubject: Subject<string>;
+    private commentFocusSubject: BehaviorSubject<{annotation: Annotation, showButton?: boolean}>;
+    private annotationFocusSubject: Subject<Annotation>;
+    private contextualToolBarOptions: Subject<{annotation: Annotation, showDelete: boolean}>;
+
+    constructor(private log: EmLoggerService,
+                private apiHttpService: AnnotationApiHttpService) {
+
+        log.setClass('AnnotationStoreService');
+        this.commentBtnSubject = new Subject();
```

Comment:
This seems weird, how come this object is instantiating it' own dependencies rather than having them passed in?